src/deluge/gui/ui/load/load_song_ui.cpp

```diff
@@ -226,233 +227,258 @@ ActionResult LoadSongUI::buttonAction(deluge::hid::Button b, bool on, bool inCar
 	return ActionResult::DEALT_WITH;
 }
 
+enum class LoadStatus { start, loadingSamples, waitingToEnd, newSongPlaying, finished };
+StorageManager* bdsm{};
```

Comment:
These globals make me feel like we should teach the task manager how to use a `std::function` instead, but it's probably fine for an initial pass

Reply:
std::function is huge to accommodate arbitrary sized captures and the type erasure indirection makes it significantly slower than function pointers. We could explore an alternate implementation with limited capture size like this one - https://www.codeproject.com/Articles/7150/Member-Function-Pointers-and-the-Fastest-Possible